### DISCUSSION TITLE
DataTable Row Expansion/ Row Group expandAll()

### DISCUSSION THREAD

**Discussion by sbasken** [primefaces/community#1570] [PrimeReact] [answered]
Hi! I have a question. We don't have `dataKey` in our data but the row expansion with row group and its toggle feature in DataTable are all working fine. However, when I try to implement expandAll(), it doesn't work with the code from the documentation since the structure of the expandedRows seems different. What we are trying to do is to have the expanded view as default when the table is rendered.

The default is I believe something like `{'id': true}` but when I log our expandedRows, it is an array of objects and each object has the entire information for each item. I think it is because we don't have id key for each object in the dataset. We have `merchantId` instead so maybe it is not being recognized? How do I implement expandAll if that is the case? I tried creating custom handle toggle function but `e.data` doesn't actually tell me which one is being clicked so I can't identify which one to toggle... Thanks in advance for your help!

**Comment by melloware** [comment, marked as answer]
I thnk you need to just follow the example and use a `dataKey` like https://primereact.org/datatable/#row_expansion   Trying to work around that to me makes no sense as its how the component was designed to work.

**Reply by sbasken** [reply]
Sorry for the delay as I had to work on other stuff for a while but thanks for the input! I tried adding a `dateKey` to it but it didn't work as expected but I think the issue maybe from not getting the unique ID for each item from the backend for the data. I'll work with the backend :) Thanks again!
